crates/telio-starcast/src/starcast_peer.rs

```diff
@@ -0,0 +1,398 @@
+//! File containing code for the starcast peer.
+
+use async_trait::async_trait;
+use boringtun::noise::{Tunn, TunnResult};
+use ipnetwork::IpNetwork;
+use std::{net::SocketAddr, time::Duration};
+use telio_crypto::{PublicKey, SecretKey};
+use telio_proto::DataMsg;
+use telio_task::{
+    io::{wait_for_tx, Chan},
+    task_exec, Runtime, RuntimeExt, Task, WaitResponse,
+};
+use telio_utils::interval;
+use telio_utils::{telio_log_error, telio_log_warn};
+use telio_wg::uapi::Peer;
+use tokio::{net::UdpSocket, sync::mpsc::error::SendTimeoutError, time::Interval};
+use x25519_dalek::{PublicKey as PublicKeyDalek, StaticSecret};
+
+/// Constant for maximum packet size.
+const MAX_PACKET: usize = 2048;
+
+#[derive(Debug, thiserror::Error)]
+/// Custom `UdpProxy` error
+pub enum Error {
+    /// `StarcastPeer` Not Configured Error
+    #[error("Starcast peer is not configured")]
+    NotConfigured,
+    /// `StarcastPeer` `IoError`
+    #[error("Starcast peer IoError")]
+    IoError(#[from] std::io::Error),
+    /// Task execution failed
+    #[error(transparent)]
+    Task(#[from] telio_task::ExecError),
+    /// IP network failed
+    #[error("Failed to parse IP network")]
+    IpNetworkError(#[from] ipnetwork::IpNetworkError),
+    /// Send failed
+    #[error(transparent)]
+    Send(#[from] SendTimeoutError<(PublicKey, DataMsg)>),
+    /// Tunnel creation failed
+    #[error("Tunnel creation failed {0}")]
+    TunnNewFail(&'static str),
+}
+
+/// The actual starcast peer containing the task which runs the wait loop for this whole component.
+pub struct StarcastPeer {
+    /// starcast virtual peer task for Telio runtime.
+    task: Task<State>,
+}
+
+/// Starcast virtual peer configuration.
+pub struct Config {
+    /// Public key of the peer with which the starcast virtual peer shall communicate.
+    public_key: PublicKey,
```

Comment:
This is the public key of the starcast virtual peer right ?

Reply:
This is the public key of libtelio.